Add a new _TestDiscovery library/target. (#981)
swiftlang/swift-testing: Documentation/ABI/TestContent.md

TODO: standardize a mechanism for third parties to produce `Test` instances
      since we don't have a public initializer for the `Test` type.
-->

## Discovering previously-emitted test content

<!--
TODO: add more detail here about how to set up a package 
-->

> To add test content discovery support to your package, add a dependency on the
> `_TestDiscovery` module in the `swift-testing` package (not the copy included
> with the Swift toolchain or Xcode), then import the module with SPI:

Reviewer: There shouldn't be copies of the module in those places anyway
```suggestion
To add test content discovery support to your package, add a dependency on the
`_TestDiscovery` module in the `swift-testing` package (it isn't available in a
Swift toolchain or from Xcode), then import the module with SPI:
```

Author: The parenthetical refers to Swift Testing, not the new module. i.e. you need the package dependency, don't go looking in Testing.framework.
